open-apparel-registry/open-apparel-registry
# Add an API parameter and UI to filter facility list items by status

## Overview

Allow facility list items to be filtered by selecting one or more statuses. It is an uncommon situation in a thousand item list to only have a dozen potential matches. Having a filter makes it dramatically easier to find and act on these items.

Connects #504 

## Demo
<img width="1395" alt="Screen Shot 2019-05-08 at 10 31 47 AM" src="https://user-images.githubusercontent.com/17363/57395303-a0d97500-717c-11e9-9680-0f46d194cfe6.png">

<img width="1396" alt="Screen Shot 2019-05-08 at 10 31 59 AM" src="https://user-images.githubusercontent.com/17363/57395313-a8008300-717c-11e9-81b0-258bb202a289.png">

<img width="1396" alt="Screen Shot 2019-05-08 at 10 32 32 AM" src="https://user-images.githubusercontent.com/17363/57395317-ab940a00-717c-11e9-86ac-ce104e9c4c17.png">


## Testing Instructions

* Reset the database 
  * `./scripts/resetdb`
  * `./scripts/processfixtures`
* Log in as `c8@example.com`
* Browse http://localhost:6543/lists/8
* Filter by different statuses and verify that the list results match what is expected.
* Filter by `POTENTIAL_MATCH` and approve one of the matches. Verify that the item disappears from the table and the item count is correctly updated.
* Approve or reject the rest of the potential matches and verify that "No matching items" is displayed in the table.
* Verify that clicking the "SHOW ALL ITEMS" button works.

## Checklist
